packages/js/components/src/select-control/index.tsx

```diff
@@ -229,6 +244,13 @@ export class SelectControl extends Component< Props, State > {
 		this.setNewValue = this.setNewValue.bind( this );
 	}
 
+	componentDidUpdate( prevProps: Props ) {
+		const { selected } = this.props;
+		if ( selected !== prevProps.selected ) {
+			this.reset( selected );
+		}
+	}
```

Comment:
Is it intended to do this for all cases? Or should we only reset the state when `virtualScroll` is `true`?

Reply:
Yes this is intended for all cases. Sorry I forgot to mention, this is a fix for initial selection where the pre-selected option is not highlighted when after a fresh reload.